fauna/models.py

```diff
@@ -16,43 +16,82 @@ def __hash__(self):
         hash(self.name)
 
 
-class DocumentReference:
-    collection: Module
-    ref_id: str
+class BaseReference:
+    _collection: Module
+    _id: str
 
-    def __init__(self, collection: Union[str, Module], ref_id: str):
-        if isinstance(collection, str):
-            self.collection = Module(collection)
+    def __init__(self, coll: Union[str, Module], id_: str):
+        if isinstance(coll, str):
+            self._collection = Module(coll)
```

Comment:
I'm not sure if the `Union` is actually checked at runtime. If not, then doing it the other way around will give a better error if a user constructs this with an `int` or something.
```suggestion
        if isinstance(coll, Module):
            self._collection = coll
```

Reply:
Oh, yeah, it's not a runtime check. I'll rework this a bit.